pages/search.html

```diff
@@ -17,22 +17,26 @@
 <ul id="search-results"></ul>
 
 <script>
-	window.data = {
-		{% for post in site.pages %}
-				{% unless post.excluded_in_search %}
-					{% if added %},{% endif %}
-					{% assign added = false %}
-					"{{ post.url | slugify }}": {
-						"id": "{{ post.url | slugify }}",
-						"title": "{{ post.title | xml_escape }}",
-						"categories": "{{ post.categories | join: ", " | xml_escape }}",
-						"url": " {{ post.url | xml_escape }}",
-						"content": {{ post.content | strip_html | replace_regex: "[\s/\n]+"," " | strip | jsonify }}
-					}
-					{% assign added = true %}
-				{% endunless %}
-		{% endfor %}
-	};
+  window.data = {
+    {% for post in site.pages %}
+      {% unless post.excluded_in_search %}
+        {% if added %},{% endif %}
+        {% assign added = false %}
+        {% assign post_url = post.url %}
+        {% if post_url != "/" %}
```

Comment:
What about the case when post.url is "/"? We would still want it to be defined as something, and not risk using the variable twice from the previous loop (if that is possible with jinja2).

Reply:
> What about the case when post.url is "/"?

Then we skip _slugify_ because it will generate the empty slug as pointed by the _Warning_ message.

```shell
Warning: Empty slug generated for '/'.
```

Rest is same as before. 😊

Reply:
My suggestion is to have an if statement that checks for the root slash, and if it’s present don’t use slugify, but default to a string like “root.” As it is now, we only generate post_url when it’s not a root slash, so the post_url for the root slash will be set to the previous value when we hit the slash. We need an else there to set that variable to something explicit.

Reply:
> but default to a string like “root.” Or something else.

No clue here why we want to do so? 🤔 And if we default to some string here, what is its use case? 
Let me know if I am missing anything here.